test/e2e_env/kubernetes/observability/applications_metrics.go

```diff
@@ -189,4 +226,41 @@ func ApplicationsMetrics() {
 		// metric from envoy
 		Expect(stdout).To(ContainSubstring("envoy_server_concurrency"))
 	})
+
+	It("should return envoy filtered metrics", func() {
+		// given
+		podName, err := PodNameOfApp(env.Cluster, "test-server-filter", namespace)
+		Expect(err).ToNot(HaveOccurred())
+		podIp, err := PodIPOfApp(env.Cluster, "test-server-filter", namespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		// when
+		stdout, _, err := env.Cluster.Exec(namespace, podName, "test-server-filter",
+			"curl", "-v", "-m", "3", "--fail", "http://"+net.JoinHostPort(podIp, "1234")+"/metrics")
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+		Expect(stdout).ToNot(BeNil())
+
+		// other application metrics
+		Expect(stdout).To(ContainSubstring("path-stats"))
+		// metric from envoy
+		Expect(stdout).To(ContainSubstring("kuma_envoy_admin"))
+
+		// when enabled usedOnly
```

Comment:
```suggestion
		// when usedOnly is enabled
```

Reply:
So, do I get this right that the first part of the test checks the functionality working without `usedOnly` and the second part with it? Can we name the test in a way that indicates that (or indicate that fact some other way)?